Cython/Coverage.py

```diff
@@ -41,6 +41,21 @@ def _find_dep_file_path(main_file, file_path, relative_path_search=False):
         rel_file_path = os.path.join(os.path.dirname(main_file), file_path)
         if os.path.exists(rel_file_path):
             abs_path = os.path.abspath(rel_file_path)
+
+        abs_no_ext = os.path.splitext(abs_path)[0]
+        file_no_ext, extension = os.path.splitext(file_path)
+        # We check if the paths match by matching the directories in reverse order.
+        # pkg/module.pyx /long/absolute_path/bla/bla/site-packages/pkg/module.c should match.
+        # this will match the pairs: module-module and pkg-pkg. After which there is nothing left to zip.
+        matching_paths = zip(reversed(abs_no_ext.split(os.sep)), reversed(file_no_ext.split(os.sep)))
```

Comment:
Should this also normalize paths?
```suggestion
        matching_paths = zip(
            reversed(map(os.path.normpath, abs_no_ext.split(os.sep))),
            reversed(map(os.path.normpath, file_no_ext.split(os.sep))),
        )
```

Reply:
I'm also not too keen on `str.split` here - I wonder if `os.path.split` would be more robust (but it'd need multiple calls to completely split a path)

Reply:
os.path.spliit is implemented using common string methods (rfind) that use the separator. So it is not necessarily more robust.